ui/components/ui/tabs/tab/tab.tsx

```diff
@@ -27,42 +26,60 @@ export const Tab = <TKey extends string = string>({
   children,
   textProps,
   disabled = false,
-  ...rest
+  ...props
 }: TabProps<TKey>) => {
-  const handleClick = (event: React.MouseEvent<HTMLLIElement>) => {
-    event.preventDefault();
+  const handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
     if (!disabled && onClick) {
       onClick(tabIndex);
     }
   };
 
   return (
     <Box
-      as="li"
       data-testid={dataTestId}
-      onClick={handleClick}
       key={tabKey}
-      {...rest}
-      className={classnames('tab', className, {
-        'tab--single': isSingleTab,
-        'tab--active': isActive,
-        'tab--disabled': disabled,
-        ...(activeClassName && isActive && { [activeClassName]: true }),
-      })}
+      role="tab"
+      tabIndex={isActive ? 0 : -1}
+      aria-selected={isActive}
+      aria-disabled={disabled}
+      {...props}
+      className={twMerge(
+        // Original class names for compatibility (e2e tests, etc.)
+        'tab',
+        isActive && 'tab--active',
+        disabled && 'tab--disabled',
+        isSingleTab && 'tab--single',
```

Comment:
These class names no longer have styles attached(in this component folder at least) but some styles are attached still used elsewhere in the app, which isn’t ideal. I previously removed them, but that caused many tests to fail, so I’ve kept them to ensure tests still pass and to limit changes in this PR.